### ISSUE TITLE
Crash in wf 20234.99, HGCal validation for Premixing test

### ISSUE THREAD

**Issue by fabiocos** [cms-sw/cmssw#26284]
After the integration of #26099 implementing the HGCalValidator, the test workflow 20234.99 (Phase 2 scenario D17, premixing) fails due to a missing collection:

```
----- Begin Fatal Exception 28-Mar-2019 09:29:04 CET-----------------------
An exception of category 'ProductNotFound' occurred while
   [0] Processing  Event run: 1 lumi: 2 event: 107 stream: 1
   [1] Running path 'validation_step6'
   [2] Calling method for module HGCalValidator/'hgcalValidator'
Exception Message:
Principal::getByToken: Found zero products matching all criteria
Looking for type: std::vector<CaloParticle>;
Looking for module label: mix
Looking for productInstanceName: MergedCaloTruth

   Additional Info:
      [a] If you wish to continue processing events after a ProductNotFound exception,
add "SkipEvent = cms.untracked.vstring('ProductNotFound')" to the "options" PSet in the configuration.

----- End Fatal Exception -------------------------------------------------
```

**Comment by fabiocos** [comment]
assign dqm

@apsallid @rovere @kpedro88 FYI

**Comment by fabiocos** [comment]
The problem looks like in the produced used in the configuration: the default is "mix"
https://github.com/cms-sw/cmssw/blob/master/Validation/HGCalValidation/python/HGCalValidator_cfi.py#L30
but the premixing puts the collection in mixData:
```
15:35 cmsdev25 705> edmDumpEventContent step3.root | grep CaloParticle
vector<CaloParticle>                  "mixData"                   "MergedCaloTruth"   "HLT"     
```
Just changing the producer let the code run smoothly

@makortel FYI

**Comment by makortel** [comment]
Right, https://github.com/cms-sw/cmssw/blob/master/Validation/HGCalValidation/python/HGCalValidator_cfi.py should have
```python
from Configuration.ProcessModifiers.premix_stage2_cff import premix_stage2
premix_stage2.toModify(hgcalValidator,
    label_cp_effic = "mixData:MergedCaloTruth",
    label_cp_fake = "mixData:MergedCaloTruth"
)
```
(and eventually we should resolve #26001)

**Comment by fabiocos** [comment]
@makortel yes, I came to the same conclusion, running a test right now

**Comment by rovere** [comment]
Thanks for promptly fixing it!

**Comment by fabiocos** [comment]
The problem looks solved by #26287 in CMSSW_10_6_X_2019-03-28-2300
